tests/foreman/virtwho/test_ui_virtwho.py

```diff
@@ -420,3 +427,201 @@ def test_positive_virtwho_configs_widget(session, form_data):
+
+    :id: 4bef3b31-89d2-4502-b971-2437b920a3ff
+
+    :expectedresults:
+        Virt-who Reporter Role granting minimal set of permissions for virt-who
+        to upload the report, it can be used if you configure virt-who manually
+        and want to use user that has locked down account.
+    """
+    username = gen_string('alpha')
+    password = gen_string('alpha')
+    config_name = gen_string('alpha')
+    with session:
+        # Create an user
+        session.user.create({
+            'user.login': username,
+            'user.mail': 'test@test.com',
+            'user.auth': 'INTERNAL',
+            'user.password': password,
+            'user.confirm': password,
+        })
+        # Create a virt-who config plugin
+        form_data['name'] = config_name
+        session.virtwho_configure.create(form_data)
+        values = session.virtwho_configure.read(config_name)
+        command = values['deploy']['command']
+        deploy_configure_by_command(command)
+        assert session.virtwho_configure.search(config_name)[0]['Status'] == 'ok'
+        # Update the virt-who config file
+        config_id = get_configure_id(config_name)
+        config_file = get_configure_file(config_id)
+        update_configure_option('rhsm_username', username, config_file)
+        delete_configure_option('rhsm_encrypted_password', config_file)
+        add_configure_option('rhsm_password', password, config_file)
+        restart_virtwho_service()
+        assert get_virtwho_status() == 'logerror'
+        # Check the permissioin of Virt-who Reporter
+        session.user.update(
+            username,
+            {'roles.resources.assigned': ['Virt-who Reporter']}
+        )
+        assert session.user.search(username)[0]['Username'] == username
+        user = session.user.read(username)
+        assert user['roles']['resources']['assigned'] == ['Virt-who Reporter']
+        restart_virtwho_service()
+        assert get_virtwho_status() == 'running'
+        with Session(test_name, username, password) as newsession:
+            assert not newsession.virtwho_configure.check_create_permission()['can_view']
+        session.user.delete(username)
+        assert not session.user.search(username)
+
+
+def test_positive_virtwho_viewer_role(session, test_name, form_data):
+    """Verify the virt-who viewer role can TRULY work.
+
+    :id: 9cdd3bbe-cfb2-4297-92fb-110515f835b3
+
+    :expectedresults:
+        Virt-who Viewer Role granting permissions to see all configurations
+        including their configuration scripts, which means viewers could still
+        deploy the virt-who instances for existing virt-who configurations.
+    """
+    username = gen_string('alpha')
+    password = gen_string('alpha')
+    config_name = gen_string('alpha')
+    with session:
+        # Create an user
+        session.user.create({
+            'user.login': username,
+            'user.mail': 'test@test.com',
```

Comment:
You can use `valid_emails_list` from robottelo.datafactory to obtain random email.
test.com seems to be a website for existing business (they offer survey product), so we probably shouldn't use it in our automation, just in case someone actually use that address to send an email.

Reply:
ok, thanks